File: src/ProjectR.Scripting/content/scripts/afflictions/Passives/lua/MagicResistant.lua
function GetName()
  return "Magic Resistant"
end

function GetType()
  return Passive
end

function GetHookPoints()
  return { TakingDamage }
end

function OnAttachment(char)
end

function OnRemoval(char)
end

function Trigger(hookPoint, char)
  if hookPoint == TakingDamage and GetCurrentSpell():GetSpellType() == Magical and Roll(0, 99) < 10 then
    param1 = 0
  end
end

File: src/ProjectR.Scripting/content/scripts/afflictions/Passives/MagicResistant.cs
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;

namespace ProjectR.Scripting.Afflictions
{
    public class MagicResistant : Affliction
    {
        public override string Name { get { return "Magic Resistant"; } }

        public override AfflictionType Type { get { return AfflictionType.Passive; } }

        protected override HookPoint[] HookPoints
        {
            get
            {
                return new[]
                {
                    HookPoint.TakingDamage, 
                };
            }
        }

        protected override void OnTakingDamage(ICharacter character, ref double damage)
        {
            if (GetCurrentSpell().SpellType == SpellType.Magical && RollPercentage(10))
            {
                damage = 0;
            }
        }
    }
}